resteasy-core/src/main/java/org/jboss/resteasy/plugins/interceptors/CacheControlFeature.java

```diff
@@ -74,4 +75,20 @@ protected CacheControl initCacheControl(NoCache value) {
             cacheControl.getNoCacheFields().add(field);
         return cacheControl;
     }
+
+    private Method findInterfaceBasedMethod(Class<?> root, Method method) {
+        if (method.getDeclaringClass().isInterface() || root.isInterface())
+            return method;
+
+        for (Class<?> intf : root.getInterfaces()) {
+            try {
+                return intf.getMethod(method.getName(), method.getParameterTypes());
+            } catch (NoSuchMethodException ignored) {
+            }
+        }
+
+        if (root.getSuperclass() == null || root.getSuperclass().equals(Object.class))
+            return method;
+        return findInterfaceBasedMethod(root.getSuperclass(), method);
+    }
```

Comment:
This isn't quite right. It should look at the method and see if it has the `@Cache` annotation before walking down the dependency chain looking for one to inherit. In other words, if the annotation is on the implementation, that should override the annotation that's on the interface method.